index.js

```diff
@@ -6,6 +6,60 @@ const KEYCODE = {
 }
 
 const state = new Map()
+// when container or children get focus
+const onFocusin = e => {
+  const {currentTarget: rover} = e
+  if (state.get('last_rover') == rover) return
+  if (state.has(rover)) {
+    activate(rover, state.get(rover).active)
+    state.set('last_rover', rover)
+  }
+}
+const onKeydown = e => {
+  const {currentTarget: rover} = e
+  switch (e.keyCode) {
+    case KEYCODE[isRtl ? 'LEFT' : 'RIGHT']:
+    case KEYCODE.DOWN:
+      e.preventDefault()
+      focusNextItem(rover)
+      break
+    case KEYCODE[isRtl ? 'RIGHT' : 'LEFT']:
+    case KEYCODE.UP:
+      e.preventDefault()
+      focusPreviousItem(rover)
+      break
+  }
+}
+const mo = new MutationObserver((mutationList, observer) =>{
+  const stateElementsSet = new Set();
+  state.forEach((v,k) =>{
+    stateElementsSet.add(k.classList[0])
+  } )
+  mutationList
+    .filter(x => x.removedNodes.length > 0)
+    .forEach(mutation => {
+      mutation.removedNodes.forEach(removedEl => {
+      if (removedEl.nodeType !== 1) return //only elements
+        state.forEach((val,key) => {
+          if (key ==='last_rover') return;
```

Comment:
i wonder if we should add `state.rovers` as the place for stashing all the rover specifics, then any other state wont need escaped (like isRtl if it gets added to state).

also seems like another good filter candidate, since we're goin functional so much?

Reply:
The Map doesn't have a filter method and I don't know if it's worth converting for what is so far 0 to 1 extra entries.